lib/FusionInventory/Agent/Tools.pm

```diff
@@ -521,16 +523,20 @@ sub runFunction {
     my $result;
     eval {
         local $SIG{ALRM} = sub { die "alarm\n" };
+
         # set a timeout if needed
-        alarm $params{timeout} if $params{timeout};
+        alarm $params{timeout} if setExpirationTime(%params);
```

Comment:
This is susceptible to fails in case of incorrect usage:
runFunction(module => 'foo', function => 'bar', expiration => 3000);
will result in a warning because params{timeout} is undefined.

The following construct seems safer:
if ($params{timeout}) {
    alarm $params{timeout};
    setExpirationTime(timeout => $params{timeout});
}

Reply:
Yes, you're right.
Thank you for your vigilance.